common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx

```diff
@@ -56,24 +61,52 @@ export default class ConfirmationModal extends React.Component {
       }
     }, 1000);
 
-    this.props.wallet.getAddress().then(address => {
-      this.setState({ address });
-    });
+    this._setWalletAddress(this.props.wallet);
   }
 
   componentWillUnmount() {
     clearTimeout(this.readTimer);
   }
 
+  _setWalletAddress(wallet: BaseWallet) {
+    wallet.getAddress().then(fromAddress => {
```

Comment:
I'll take some blame for this, since I 👍 the proposal to make it this way. But I agree, after having worked with wallets a bit more, it seems like all of this should get resolved during unlock. Apologies for my misguided earlier approval.